src/python/grpcio/grpc/__init__.py

```diff
@@ -1744,6 +1744,44 @@ def dynamic_ssl_server_credentials(initial_certificate_configuration,
             certificate_configuration_fetcher, require_client_authentication))
 
 
+@enum.unique
+class LocalConnectionType(enum.Enum):
+    """Type of local connections for which local channel/server credentials will be applied.
+
+    Attributes:
+      UDS: Unix domain socket connections
+      LOCAL_TCP: Local TCP connections.
+    """
+    UDS = _cygrpc.LocalConnectType.uds
+    LOCAL_TCP = _cygrpc.LocalConnectType.local_tcp
+
+
+def local_channel_credentials(local_connect_type=LocalConnectionType.LOCAL_TCP):
+    """Creates a local ChannelCredentials used for local connections.
```

Comment:
I think we're going to want a longer docstring describing what local credentials are and perhaps naming potential use cases. Without context, this won't be useful to very many users.

Reply:
Huh. Apparently, [this is still considered an experimental API](https://github.com/grpc/grpc/blob/13fb5066390b51bf457725a91fe3a93daed40940/include/grpc/grpc_security.h#L687) within core. At any rate, we probably want some explanation about exactly what sort of authentication/authorization mechanism is used. Without looking into the code, even I'm still scratching my head about what exactly this feature is.

Reply:
Thanks Lidi for exposing the local credential in OSS. 

The current status is C-core, Java and Go teams agreed on the semantics of channel and call credentials (I shared with you and Lidi) which will be a building block for designing local credentials in all languages. I still need to update the existing API's in those three languages to conform to the semantics (Sorry for taking so long), and after that we can design local credentials for Java and Go and remove experimental namespace in C-core. 

W.r.t explanation, TCP loopback does not provide either peer authentication or channel confidentiality while UDS provides both of them and can be treated as secure as transport security.

Reply:
@gnossen  PTAL at the added comments.
@yihuazhang I added this because in the past users is complaining that API disparity prevents them from using `secure_channel`.